playbooks/provisioning/openstack/prerequisites.yml

```diff
@@ -85,6 +87,20 @@
       that: 'key_result.rc == 0'
       msg: "Keypair {{ openstack_ssh_public_key }} is not available"
 
+  - name: The provider and private networks can't be set at the same time
+    fail:
+      msg: You cannot set both `openstack_provider_network_name` and `openstack_private_network_name` at the same time
+    when:
+    - openstack_provider_network_name|default(None)
+    - openstack_private_network_name|default(None)
+
+  - name: The provider and external networks can't be set at the same time
+    fail:
+      msg: You cannot set both `openstack_provider_network_name` and `openstack_external_network_name` at the same time
+    when:
+    - openstack_provider_network_name|default(None)
+    - openstack_external_network_name|default(None)
```

Comment:
ditto. No need to fail, IMO, let it be ignored instead. Assuming users know that they want, when defining a provider network name.